wire-ios/WireCommonComponents/Analytics/WireAnalytics.swift

```diff
@@ -39,16 +40,71 @@ public enum WireAnalytics {
 
         WireAnalytics.Datadog.shared.enable()
 
+        migrateLogFilesToNewLocation(target: target)
+
         WireLogger.initialize {
             #if DEBUG
                 SystemLogger()
             #endif
-            CocoaLumberjackLogger()
+            CocoaLumberjackLogger(logsDirectory: logsDirectory(for: target))
             WireAnalytics.Datadog.shared
         }
 
         // pass tags to Datadog through WireLogger
         WireLogger.system.addTag(.processId, value: "\(ProcessInfo.processInfo.processIdentifier)")
         WireLogger.system.addTag(.processName, value: ProcessInfo.processInfo.processName)
     }
+
+    private static func logsDirectory(for target: LogTarget) -> URL? {
+        guard let appGroupIdentifier = Bundle.main.applicationGroupIdentifier else { return nil }
+        return FileManager.default.sharedLogsDirectoryURL(for: appGroupIdentifier)?
+            .appending(component: target.rawValue, directoryHint: .isDirectory)
+    }
+
+    /// With release 4.2.0 the log files of all targets (main app, notification service extension and share extension)
+    /// are written to the shared container. This code tries to move the old files for the case that all log files want
+    /// to be retrieved/exported.
+    /// Eventually this func can just be deleted.
+
+    private static func migrateLogFilesToNewLocation(target: LogTarget) {
+        do {
+
+            let fileManager = FileManager.default
+            guard let newRootDirectory = logsDirectory(for: target) else { return }
+            let previousRootDirectory = try fileManager.url(
+                for: .cachesDirectory,
+                in: .userDomainMask,
+                appropriateFor: nil,
+                create: false
+            ).appending(path: "Logs", directoryHint: .isDirectory)
+
+            guard fileManager.fileExists(atPath: previousRootDirectory.path()) else { return }
+
+            let previousLogFileURLs = try fileManager.contentsOfDirectory(
+                at: previousRootDirectory,
+                includingPropertiesForKeys: nil
+            )
+            if !previousLogFileURLs.isEmpty {
+                try fileManager.createDirectory(at: newRootDirectory, withIntermediateDirectories: true)
+            }
+
+            for previousLogFileURL in previousLogFileURLs {
+                let newLogFileURL = newRootDirectory.appending(
+                    path: previousLogFileURL.lastPathComponent,
+                    directoryHint: .notDirectory
+                )
+                if fileManager.fileExists(atPath: newLogFileURL.path()) {
+                    continue // don't overwrite, instead delete the old file (unlikely anyway)
```

Comment:
question: as you update, you'll start writing in the new location, even before you migrate old files, no?

we can simply move the old file in the new location to be zipped and let the new file be zipped when we send